src/main/java/org/apache/sysds/runtime/compress/estim/encoding/EncodingFactory.java

```diff
@@ -253,9 +303,28 @@ private static IEncode createFromDense(MatrixBlock m, int col) {
 		final int end = off + nRow * nCol;
 		final double[] vals = m.getDenseBlockValues();
 
+		// Validate scaleFactors
+		boolean useSingleScalar = false;
+		if(scaleFactors != null) {
+			if(scaleFactors.length == 1) {
+				useSingleScalar = true;
+			}
+		}
+
 		// Iteration 1, make Count HashMap.
-		for(int i = off; i < end; i += nCol) // jump down through rows.
-			map.increment(vals[i]);
+		for(int i = off, r = 0; i < end; i += nCol, r++) {// jump down through rows.
+			double value = vals[i];
```

Comment:
these modification of inner for loops slow things down, please branch out before the for loop and make two versions, one with scaling one without.